controller/tenant.go

```diff
@@ -337,6 +298,74 @@ func listNames(namespaces []*tenant.Namespace) []string {
 	return names
 }
 
+func (c *TenantController) getOrInitTenant(ctx context.Context, user *auth.User) (*tenant.Tenant, error) {
+	var dbTenant *tenant.Tenant
+	var err error
+	tenantRepository := c.tenantService.NewTenantRepository(user.ID)
```

Comment:
why do you pass the `user.ID` in the tenant repository constructor? shouldn't it be user-agnostic (hence, pass the `user.ID` as an arg when calling the methods. eg: `repository.Exists(user.ID)`)

Reply:
Because the `TenantRepository` is specific for one tenant. In one request I'm dealing with one tenant and this helps me to minimize the number of parameters passed to other methods/objects - I don't have to pass DB service along with the tenant id because it is already contained within the repository: https://github.com/fabric8-services/fabric8-tenant/blob/develop/tenant/repository.go#L138

Reply:
yes, that's what I understood, but it feels unusual to construct the repository with the tenant associated to it. My point is that it does not follow our pattern for services and repositories.

Reply:
For me it makes sense - it represents a repository for one tenant entity comparing to the `DBService` that represents a service for all other cases (not related to one tenant). 
But I'm open to any proposal ;-).
we already had some discussions about it:
https://github.com/fabric8-services/fabric8-tenant/pull/673#discussion_r248610738
https://github.com/fabric8-services/fabric8-tenant/pull/732#discussion_r249411930

I have a big problem with these namings so I would welcome any help :-)